docs/04-reference/winglang-spec.md

```diff
@@ -1147,8 +1149,8 @@ resource Foo {
   field6: bool;
 
   // re-assignable class fields, read about them in the mutability section
-  readwrite field4: num;
-  readwrite field5: str;
+  var field4: num;
```

Comment:
The following suggestion came up, instead of `var` change this to `let var`. And in class definitions do:
```wing
class {
  x: num;
  var y: num; // I'm reassignable
}
```
This creates some consistency where `var` is a modifier on the symbol meaning it's reassignable. It also adds some "ugliness" or friction meant to discourage the use of `var`. 
We should also add a separate issue for adding a compiler warning if you create a `var` and never reassign to it.

Reply:
+1 on 

```js
let var x = 12;
```

Reply:
By the way, tangentially related: when you declare a var in a class, you either must define its value in the ctor (static analysis) or it must be optional (and then we go into the `nil` world)

Reply:
> By the way, tangentially related: when you declare a var in a class, you either must define its value in the ctor (static analysis) or it must be optional (and then we go into the nil world)

In general I'm for either having default values for **everything** optional or not or forcing everything to be inited in the ctor. In the latter case, even an optional needs to be inited: `init() {self.my_optional_num = nil; // or undefined or nothing...}`
In the former case we can have strict rules saying `num`s default to 0, `str`s default to "", etc.

Reply:
What about `let mut` (and `mut` for fields), following Rust?

Reply:
We are intentionally separating between what we call "mutable" and "reassignable", which I believe is a source of confusion in Rust.

In the following example, `x` *cannot be reassigned*, but it is *mutable* (there's a way to mutate it's value).

```js
let x = new MyObject();
x.foo = 12;
```

So we don't want to use the term `mut` to represent reassinability.

Reply:
Updated the syntax to `let var`